### ISSUE TITLE
unix soft link problems

### ISSUE THREAD

**Issue by aristotll** [xylo/intellij-postfix-templates#53]
I use soft link to ensure the portability of my templates.
But it does not work.

- If I use soft link for the whole templates dir, I have to restart  to update the templates even if I have pressed ctrl s(which is command s in mac osx) to save the edited files.

- If I use soft link for specific file such as `java.postfixTemplates`

```
templates ll
total 32
lrwxr-xr-x  1 Yao  staff   107B  2 10 17:38 java.postfixTemplates -> /Users/Yao/codegit/Java/JavaRTDemo/src/main/java/velocity/templates/postfix/templates/java.postfixTemplates
```

The templates in file `java.postfixTemplates` cannot even be loaded.

**Comment by xylo** [comment]
I guess I know what's the problem.

I currently use IDEA's file change notification mechanism to reload the templates when I file changes.  However, this mechanism is probably using inotify in background and I guess it's not following symlinks.  This would explain that the templates don't get reloaded when you press Ctrl+S.

So I probably have to add the old template reload mechanism again which was originally triggered by Ctrl+S.

**Comment by xylo** [comment]
Fixed in upcoming version 1.6.5.

**Comment by aristotll** [comment]
I am afraid the problem is not fixed, at least in my Mac.

**Comment by aristotll** [comment]
<img width="790" alt="2018-04-22 9 07 47" src="https://user-images.githubusercontent.com/13341544/39095330-506d5ec4-4671-11e8-8060-bd873e8b37e8.png">
<img width="529" alt="2018-04-22 9 07 53" src="https://user-images.githubusercontent.com/13341544/39095331-50ac54e4-4671-11e8-8abc-b3e22e0e7993.png">

**Comment by xylo** [comment]
Hm. OK.

**Comment by xylo** [comment]
Indeed, I was to optimistic that it would work with the old algorithm, but it does not.  I guess that IDEA resolves the path of the link and saves the file under it's real path, so that the plugin does not detect the file save operation.  I have to see whether I can resolve the real path of the link as well and check for changes of the real file.

Moreover this will be anyway fixed in version 2, because v2 will allow you to select arbitrary files from the file system, so that you don't have to use symlinks anymore.

**Comment by aristotll** [comment]
Thanks for your quick response, also great new for v2.

By the way how to write new line in templates? I try adding a trailing `\` but it seems that is not working.

**Comment by xylo** [comment]
Yeah. Sorry, I broke the newline escapes at some point, but I fixed it yesterday in version 1.8.1

**Comment by xylo** [comment]
Version 2.0.0 and later versions support local template files which solves the problem IMO.  So instead of creating symlinks/links you can link to your files using "local template files".
